### ISSUE TITLE
master doesn't pass tox tests - AttributeError: type object 'Function' has no attribute 'from_parent'

### ISSUE THREAD

**Issue by panasenco** [robinhood/faust#603]
## Checklist

- [x] I have included information about relevant versions: **1.11.0a1**
- [x] I have verified that the issue persists when using the `master` branch of Faust: **commit 8b22767**

## Steps to reproduce

Run `tox -e 3.8`

## Expected behavior

All tests pass

## Actual behavior

tox unable to even start tests

## Full traceback

```
faust> git pull --rebase upstream master
From git://github.com/robinhood/faust
 * branch              master     -> FETCH_HEAD
Already up to date.
Current branch master is up to date.
faust> tox -e 3.8
GLOB sdist-make: C:\Users\apanasenco\git\faust\setup.py
3.8 inst-nodeps: C:\Users\apanasenco\git\faust\.tox\.tmp\package\1\faust-1.11.0a1.zip
3.8 installed: aiofiles==0.5.0,aiohttp==3.6.2,aiohttp-cors==0.7.0,aredis==1.1.8,async-timeout==3.0.1,atomicwrites==1.4.0,attrs==19.3.0,certifi==2020.4.5.1,chardet==3.0.4,click==7.1.2,codecov==2.1.3,colorama==0.4.3,colorclass==2.2.0,colorlog==4.1.0,coverage==5.1,croniter==0.3.32,datadog==0.36.0,decorator==4.4.2,faust @ file:///C:/Users/apanasenco/git/faust/.tox/.tmp/package/1/faust-1.11.0a1.zip,freezegun==0.3.15,hypothesis==5.16.0,idna==2.9,kafka-python==1.4.7,mode==4.3.2,more-itertools==8.3.0,multidict==4.7.6,mypy==0.770,mypy-extensions==0.4.3,natsort==7.0.1,opentracing==1.3.0,packaging==20.4,pluggy==0.13.1,psutil==5.7.0,py==1.8.1,pyfakefs==4.0.2,pyparsing==2.4.7,pytest==5.2.4,pytest-aiofiles==0.2.0,pytest-aiohttp==0.3.0,pytest-asyncio==0.12.0,pytest-cov==2.9.0,pytest-forked==1.1.3,pytest-openfiles==0.5.0,pytest-picked==0.4.4,pytest-random-order==1.0.4,python-dateutil==2.8.1,pytz==2020.1,PyYAML==5.3.1,requests==2.23.0,robinhood-aiokafka==1.1.6,six==1.15.0,sortedcontainers==2.1.0,statsd==3.3.0,terminaltables==3.1.0,typed-ast==1.4.1,typing-extensions==3.7.4.2,urllib3==1.25.9,venusian==1.2.0,wcwidth==0.1.9,yarl==1.4.2
3.8 run-test-pre: PYTHONHASHSEED='901'
3.8 run-test: commands[0] | py.test --random-order --open-files -xvv --cov=faust t/unit t/functional t/integration t/meticulous/ t/regression
========================================================= test session starts ========================================================= platform win32 -- Python 3.8.3, pytest-5.2.4, py-1.8.1, pluggy-0.13.1 -- c:\users\apanasenco\git\faust\.tox\3.8\scripts\python.exe
cachedir: .tox\3.8\.pytest_cache
hypothesis profile 'default' -> database=DirectoryBasedExampleDatabase('C:\\Users\\apanasenco\\git\\faust\\.hypothesis\\examples')
Using --random-order-bucket=module
Using --random-order-seed=269783

rootdir: C:\Users\apanasenco\git\faust, inifile: setup.cfg
plugins: hypothesis-5.16.0, pyfakefs-4.0.2, aiofiles-0.2.0, aiohttp-0.3.0, asyncio-0.12.0, cov-2.9.0, forked-1.1.3, openfiles-0.5.0, picked-0.4.4, random-order-1.0.4
collected 10 items / 1 errors / 9 selected

=============================================================== ERRORS ================================================================ _______________________________________________ ERROR collecting t/unit/test_events.py ________________________________________________ .tox\3.8\lib\site-packages\pluggy\hooks.py:286: in __call__
    return self._hookexec(self, self.get_hookimpls(), kwargs)
.tox\3.8\lib\site-packages\pluggy\manager.py:93: in _hookexec
    return self._inner_hookexec(hook, methods, kwargs)
.tox\3.8\lib\site-packages\pluggy\manager.py:84: in <lambda>
    self._inner_hookexec = lambda hook, methods, kwargs: hook.multicall(
.tox\3.8\lib\site-packages\_pytest\python.py:202: in pytest_pycollect_makeitem
    res = outcome.get_result()
.tox\3.8\lib\site-packages\pytest_asyncio\plugin.py:39: in pytest_pycollect_makeitem
    item = pytest.Function.from_parent(collector, name=name)
E   AttributeError: type object 'Function' has no attribute 'from_parent'
========================================================== 1 error in 2.29s =========================================================== ERROR: InvocationError for command 'C:\Users\apanasenco\git\faust\.tox\3.8\Scripts\py.test.EXE' --random-order --open-files -xvv --cov=faust t/unit t/functional t/integration t/meticulous/ t/regression (exited with code 1)
_______________________________________________________________ summary _______________________________________________________________ ERROR:   3.8: commands failed
faust> git log -1
commit 8b22767f0ccbcfb1705a8880c6b72d33924d96ca (HEAD -> master, upstream/master, origin/master, origin/HEAD)
Author: Ask Solem <asksol@robinhood.com>
Date:   Wed Apr 29 15:50:21 2020 -0700

    Fixes typecheck;
```

# Versions

* Python version: 3.8.3
* Faust version: 1.11.0a1
* Operating system: Windows 10
* Kafka version: N/A (do tox tests need a running Kafka instance?)
* RocksDB version (if applicable)

**Comment by deed02392** [comment]
Does tox for 3.7 pass?

**Comment by g-clef** [comment]
I chased this down: it's an incompatibility between `pytest` and `pytest-asyncio` versions. In `requirements/test.txt` `pytest` is pinned to  `~=5.2.2`, but `pytest-asyncio` is `>=0.8`. That means `pytest-asyncio` moved onto a version that isn't compatible with pytest 5.2.

If you change `requirements/tests.txt` to have `pytest>=5.4.0` tox will run. Some of the tests fail for me locally, but tox does at least start. (I had to clear out the .tox folder entirely for that change to take hold...otherwise the older version of pytest kept being used.)
